File: internal/models/vaccine.go
package models

import (
	"time"
)

type Vaccine struct {
	ID          uint      `gorm:"primaryKey"`
	FarmID     uint      `gorm:"not null"`
	Farm       Farm      `gorm:"foreignKey:FarmID"`
	Name       string    `gorm:"not null"`
	Description string
	Manufacturer string
	CreatedAt  time.Time
	UpdatedAt  time.Time
}

File: src/pages/contents/AnimalTable/domain/model/vaccine.ts
export interface Vaccine {
  id: number
  farm_id: number
  name: string
  description?: string
  manufacturer?: string
  created_at: string
  updated_at: string
}

export interface CreateVaccineRequest {
  farm_id: number
  name: string
  description?: string
  manufacturer?: string
}

export interface UpdateVaccineRequest {
  farm_id: number
  name: string
  description?: string
  manufacturer?: string
}

